src/AsmResolver.DotNet/Code/Cil/CilMethodBody.cs

```diff
@@ -13,6 +15,256 @@ namespace AsmResolver.DotNet.Code.Cil
     /// </summary>
     public class CilMethodBody : MethodBody, ICilOperandResolver
     {
+         /// <summary>
+        ///     Creates a CIL method body from a dynamic method.
+        /// </summary>
+        /// <param name="method">The method that owns the method body.</param>
+        /// <param name="dynamicMethodObj">The dynamic method.</param>
+        /// <param name="operandResolver">
+        ///     The object instance to use for resolving operands of an instruction in the
+        ///     method body.
+        /// </param>
+        /// <returns>The method body.</returns>
+        public static CilMethodBody FromDynamicMethod(MethodDefinition method, object dynamicMethodObj,
+            ICilOperandResolver operandResolver = null,ReferenceImporter importer = null)
+        {
+            var result = new CilMethodBody(method);
+
+            if (operandResolver is null)
+                operandResolver = result;
+
+            if(importer is null) 
+                importer = new ReferenceImporter(method.Module);
+
+
+            //Get dynamic method
+            if (dynamicMethodObj is Delegate deleg)
+                dynamicMethodObj = deleg.Method;
+
+            if (dynamicMethodObj is null)
+                throw new ArgumentNullException(nameof(dynamicMethodObj));
+
+            if (dynamicMethodObj.GetType().FullName.Contains("RTDynamicMethod"))

```

Comment:
`GetType().FullName.Contains` or `GetType().ToString().Contains` to check if the object is of type is very unreliable.